### PR TITLE
Vector processing issue

### PR DESCRIPTION
**Pull request by alalek**
Problem is in layout mismatch of `src` and `dst` for some cases:
```
OutputArray _dst(dst1);
{
    _dst.create(src.rows, src.cols, src.type());
    Mat dst = _dst.getMat();
    EXPECT_EQ(src.dims, dst.dims);
    EXPECT_EQ(src.cols, dst.cols);
    EXPECT_EQ(src.rows, dst.rows);
}
```

Only `copyTo()` memory corruption problem was fixed.

### CODE REVIEW

**Comment by alalek** [comment]
@vpisarev Vadim, could you take a look on this issue?

Are there any layout policy to `std::vector`? Should it be "row vector" or "column-vector" (or may be used somehow in both cases, that actually doesn't work well)?

**Comment by vpisarev** [comment]
initially vector was meant as Nx1 matrix (single column-vector). However later I found quite a bit of code where vector is treated as single-row matrix. So now we just try not to break any function. And of course it's not only about OpenCV itself, but also about the user code.

File: modules/core/src/copy.cpp [hunk 1]
```diff
@@ -285,6 +285,8 @@ void Mat::copyTo( OutputArray _dst ) const
 
         if( rows > 0 && cols > 0 )
         {
+            if (_dst.isVector())
+                dst = dst.reshape(0, dst.cols);
```
Comment (vpisarev) [inline modules/core/src/copy.cpp]: such call of reshape gives valid results as long as dst is 1xN before the call.
I suggest to change it to more robust
<pre>
dst = dst.reshape(0, dst.cols*dst.rows);
</pre>

**Comment by vpisarev** [comment]
:+1:

**Pull request merged**